server/src/main/java/com/broadcom/lsp/cobol/core/visitor/VariableUsageDelegate.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package com.broadcom.lsp.cobol.core.visitor;
+
+import com.broadcom.lsp.cobol.core.CobolParser;
+import com.broadcom.lsp.cobol.core.messages.MessageService;
+import com.broadcom.lsp.cobol.core.model.ErrorSeverity;
+import com.broadcom.lsp.cobol.core.model.Locality;
+import com.broadcom.lsp.cobol.core.model.SyntaxError;
+import com.broadcom.lsp.cobol.core.model.variables.Variable;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.RuleContext;
+import org.antlr.v4.runtime.Token;
+
+import java.util.*;
+
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * This class process the variable usage contexts. It accumulate usages when they appear in the text. Later it checks
+ * variable definition and updates its usages.
+ */
+@Slf4j
+@RequiredArgsConstructor
+class VariableUsageDelegate {
+  private List<VariableUsage> variableUsages = new ArrayList<>();
+  private final Map<Token, Locality> positionMapping;
+  private final MessageService messageService;
+
+  /**
+   * Accumulates variable appearance for analise its definition later.
```

Comment:
By convention, method javadocs should begin with a verb in the infinitive form, active voice.